src/OpenTelemetry.AutoInstrumentation.Native/cor_profiler.cpp

```diff
@@ -1770,6 +1778,472 @@ HRESULT CorProfiler::RunAutoInstrumentationLoader(const ComPtr<IMetaDataEmit2>&
+        if (FAILED(hr))
+        {
+            Logger::Warn("GenerateAppDomainAssemblyLoaderMethod: Failed to define assembly resolver method");
+            return hr;
+        }
+    }
+
+    mdString auto_instrumentation_assembly_name;
+    {
+        hr = metadata_emit->DefineUserString(managed_profiler_full_assembly_version_strong_name.c_str(),
+                                             (ULONG)managed_profiler_full_assembly_version_strong_name.length(),
+                                             &auto_instrumentation_assembly_name);
+
+        if (FAILED(hr))
+        {
+            Logger::Warn("GenerateAppDomainAssemblyLoaderMethod: Failed to define assembly name string");
+            return hr;
+        }
+    }
+
+    mdString auto_instrumentation_assembly_path;
+    {
+        LPCWSTR auto_instrumentation_assembly_path_str      = otel_assembly_path.native().c_str();
+        auto    auto_instrumentation_assembly_path_str_size = wcslen(auto_instrumentation_assembly_path_str);
+
+        hr = metadata_emit->DefineUserString(auto_instrumentation_assembly_path_str,
+                                             (ULONG)auto_instrumentation_assembly_path_str_size,
+                                             &auto_instrumentation_assembly_path);
+
+        if (FAILED(hr))
+        {
+            Logger::Warn("GenerateAppDomainAssemblyLoaderMethod: Failed to define assembly path string");
+            return hr;
+        }
+    }
+
+    {
+        // Generate a locals signature defined in the following way:
+        //   [0] class System.Reflection.Assembly
+        mdSignature      assembly_resolver_locals_signature_token;
+        SignatureBuilder assembly_resolver_local_signature =
+            SignatureBuilder{IMAGE_CEE_CS_CALLCONV_LOCAL_SIG, 1, ELEMENT_TYPE_CLASS}.PushToken(assembly_def);
+
+        hr = metadata_emit->GetTokenFromSig(assembly_resolver_local_signature.Head(),
+                                            assembly_resolver_local_signature.Size(),
+                                            &assembly_resolver_locals_signature_token);
+        if (FAILED(hr))
+        {
+            Logger::Warn("GenerateAppDomainAssemblyLoaderMethod: Unable to generate locals signature. ModuleID=",
+                         module_id);
+            return hr;
+        }
+
+        ILRewriter rewriter_assembly_resolver(this->info_, nullptr, module_id, assembly_resolver_def);
+        rewriter_assembly_resolver.InitializeTiny();
+        rewriter_assembly_resolver.SetTkLocalVarSig(assembly_resolver_locals_signature_token);
+
+        ILInstr* pLastInstr = rewriter_assembly_resolver.GetILList()->m_pNext;
+        ILInstr* pNewInstr  = nullptr;
+
+        std::vector<ILInstr*> branches;
+        EHClause              eh_clause{};
+        eh_clause.m_Flags      = COR_ILEXCEPTION_CLAUSE_NONE;
+        eh_clause.m_ClassToken = exception_def;
+
+        pNewInstr           = rewriter_assembly_resolver.NewILInstr();
```

Comment:
could you add also short comments for each block, what is the aim.

examples in cor_profiler.cpp, LN 1909 and further. Looks like this:

```
    // ldsflda _isAssemblyLoaded : Load the address of the "_isAssemblyLoaded" static var
    pALNewInstr           = rewriter_already_loaded.NewILInstr();
    pALNewInstr->m_opcode = CEE_LDSFLDA;
    pALNewInstr->m_Arg32  = isAssemblyLoadedFieldToken;
    rewriter_already_loaded.InsertBefore(pALFirstInstr, pALNewInstr);
```

Reply:
I've added comments at the begin of function with C# code that was used to get IL and before each IL segment with raw IL code. Probably it will be easier to understand that way than with comment before each IL command.
Please check and advice if still each IL line comment still required.